src/operator/controllers/protected_service_reconcilers/default_deny.go

```diff
@@ -67,8 +68,15 @@ func (r *DefaultDenyReconciler) blockAccessToServices(ctx context.Context, prote
 			continue
 		}
 
-		formattedServerName := otterizev1alpha3.GetFormattedOtterizeIdentity(protectedService.Spec.Name, namespace)
-		policy := r.buildNetworkPolicyObjectForIntent(formattedServerName, protectedService.Spec.Name, namespace)
+		serverServiceIdentity := protectedService.ToServiceIdentity()
+		formattedServerName := serverServiceIdentity.GetFormattedOtterizeIdentityWithKind()
```

Comment:
Now that this can refer to a `kind: Service` by its labels, does it also get called when the `Service` changes? I couldn't see that it is

Reply:
I couldn't find a test for this, can you add one in default_deny_test.go?